# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/test/zoom/head.js
# Commit: febdcf7986a3
# Full Hash: febdcf7986a37651188476cd8253d20e9cea443c
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2022-06-23 03:54:50
# Regressor Bug: 1773865
# File Overlap Count: 1
# Description:
#   Bug 1773865 - Dispatch an event on the window document when a pinch zoom gesture ends. r=botond,smaug,NeilDeakin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149283
# ==============================================================================

diff -r 00d1c650d88f -r febdcf7986a3 browser/base/content/test/zoom/head.js
--- a/browser/base/content/test/zoom/head.js	Wed Jun 22 19:26:58 2022 +0000
+++ b/browser/base/content/test/zoom/head.js	Wed Jun 22 19:40:17 2022 +0000
@@ -32,12 +32,7 @@
      */
 
     let parsedZoomValue = parseFloat((parseInt(newZoom) / 100).toFixed(2));
-
-    let commandsUpdated = BrowserTestUtils.waitForEvent(
-      window,
-      "ZoomCommandsUpdated"
-    );
-    let completion = new Promise(resolve => {
+    await new Promise(resolve => {
       gContentPrefs.setGlobal(
         FullZoom.name,
         parsedZoomValue,
@@ -49,7 +44,16 @@
         }
       );
     });
-    await Promise.all([commandsUpdated, completion]);
+    // The zoom level is used to update the commands associated with
+    // increasing, decreasing or resetting the Zoom levels. There are
+    // a series of async things we need to wait for (writing the content
+    // pref to the database, and then reading that content pref back out
+    // again and reacting to it), so waiting for the zoom level to reach
+    // the expected level is actually simplest to make sure we're okay to
+    // proceed.
+    await TestUtils.waitForCondition(() => {
+      return ZoomManager.zoom == parsedZoomValue;
+    });
   },
 
   async getGlobalValue() {